src/domain/shared/primitives.ts

```diff
@@ -92,6 +92,19 @@ export const paymentAmountFromNumber = <T extends WalletCurrency>({
   }
 }
 
+export const displayAmountFromNumber = <T extends DisplayCurrency>({
+  amount,
+  currency,
+}: {
+  amount: number
+  currency: T
+}): DisplayAmount<T> | ValidationError => {
```

Comment:
is `ValidationError` needed?

Reply:
Ah with the current implementation no, but I plan to change these to bigints and have a similar interface as `paymentAmountFromNumber` in a follow-up PR... can revert this change for now if we think we should